### PR TITLE
Feature/ms/counselor and clients

### PR DESCRIPTION
**Pull request by nailaghigo**
Pull request to add:

Path to Clients and Counselors --> index.js file.
Controllers for Clients and Counselors --> clients.js and counselors.js inside controllers folder.
Data to test Clients and Counselors --> Json files inside data folder

### CODE REVIEW

File: controllers/clients.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    }
+    return res.json(filterClients);
+}
+
+// Get Client by ID
+const getClientsById = (req, res) => {
+    const id = req.params.id;
+    let filterClientsId = clients.filter(item => item.id == id)
+    if (filterClientsId.length === 0){
+        return res.status(400).json({ msg: `Client with ID ${id} not found`});
```
Comment (aquilesmariani) [inline controllers/clients.js:27]: The status code for when the requested resource hasn't been found is `404`.

Also, there's no need to return a message as a json format inside of an object. We could just a string instead:

```suggestion
        return res.status(404).send(`Client with ID ${id} not found`);
```

**Review by aquilesmariani — COMMENTED** [review]
There are some repetitive minor mistakes, just make those adjustments and it will be ready to go!

File: controllers/clients.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        email: email,
+        phone: req.query.phone,
+        open_positions: [],
+    }
+    clients.push(newClient);
+    fs.writeFile('./data/clients.json', JSON.stringify(clients), (err) => {
+        if (err) {
+            return res.status(400).json({ msg: 'Couldn\'t create Client file'});
+        } else {
+            return res.status(200).json({ msg: 'New Client succesfully created'});
```
[the remark this answers is not in the document]
Comment (nailaghigo) [reply controllers/clients.js]: You're right! changes done on counselors.js and clients.js!

File: controllers/clients.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
+const clients = require('../data/clients.json');
+const fs = require('fs');
+
+// Get all Clients
+const getClients = (req, res) => {
+    if (clients.length === 0){
+        return res.status(400).json({ msg: 'Clients not found'});
```
[the remark this answers is not in the document]
Comment (nailaghigo) [reply controllers/clients.js]: if conditional removed!

File: controllers/clients.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    }
+    return res.json(clients);
+}
+
+// Get Clients by Company Type
+const getClientsByCompanyType = (req, res) => {
+    const companyType = req.params.company_type;
+    let filterClients = clients.filter(item => item.company_type.toLowerCase().includes(companyType.toLowerCase()))
+    if (filterClients.length === 0){
+        return res.status(400).json({ msg: `Clients from company type ${companyType} not found`});
```
[the remark this answers is not in the document]
Comment (nailaghigo) [reply controllers/clients.js]: if conditional removed!

File: controllers/clients.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
+const clients = require('../data/clients.json');
+const fs = require('fs');
+
+// Get all Clients
```
[the remark this answers is not in the document]
Comment (nailaghigo) [reply controllers/clients.js]: Got it! All those comments are deleted now.Thansk a lot for your suggestion!

File: controllers/clients.js (hunk 1 above)
Comment (nailaghigo) [reply controllers/clients.js]: Done!

**Comment by nailaghigo** [comment]
@aquilesmariani All Changes Done, Please let me know if they are OK to Merge

**Review by aquilesmariani — APPROVED** [review]
Looks great! Let's merge this one now!

**Pull request merged**